src/test/run-make/debug-assertions/debug.rs

```diff
@@ -0,0 +1,22 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![deny(warnings)]
+
+fn main() {
+    let mut hit1 = false;
+    let mut hit2 = false;
+    debug_assert_eq!({ hit1 = true; 1 }, { hit2 = true; 2 });
+    assert!(!hit1);
+    assert!(!hit2);
+    let mut hit = false;
+    debug_assert!({ hit = true; false });
```

Comment:
hmm, how will you ever reach this line?  If debug-assertion are on, won't the earlier `debug_assert_eq` fail?

If your intent is to test the correct behavior of side-effects (or lack thereof) that occur within `debug_assert` and `debug_assert_eq`, I think you need to make separate tests.